### PR TITLE
Added queue for pending Start/Stop forging commands consequently the …

### PR DESCRIPTION
**Pull request by alzinchenko**
Added queue for pending Start/Stop forging commands so the API startForging/stopForging methods responses immediately;

### CODE REVIEW

**Review by yuriy-larin-firstbridge — APPROVED** [review]

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java [hunk 1]
```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
...
-            globalSync.updateUnlock();
-        }
-        log.debug("stopped forging ({} ms)", (System.currentTimeMillis() - start));
+        log.debug("Admin: Stop Forging command added to the queue.");
+        setConditionToResetForgers();
         return count;
     }
 
+    private void setConditionToResetForgers() {
+        if (generateBlocksTask != null) {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:147]: Probably the check: `!propertiesHolder.isLightClient()` could bring more clarity here

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java [hunk 2]
```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
...
-        try {
-            if (generateBlocksTask != null) {
-                generateBlocksTask.resetSortedForgers();
-            }
-        } finally {
-            globalSync.updateUnlock();
-        }
-        log.debug("stopped forging ({} ms)", (System.currentTimeMillis() - start));
+        log.debug("Admin: Stop Forging command added to the queue.");
+        setConditionToResetForgers();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:142]: Maybe better to name it: scheduleAsyncForgersReset or smth like that

Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:142]: renamed

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java (hunk 1 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:147]: added

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java [hunk 3]
```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
...
-            globalSync.updateUnlock();
-        }
-        log.debug("stopped forging ({} ms)", (System.currentTimeMillis() - start));
+        log.debug("Admin: Stop Forging command added to the queue.");
+        scheduleAsyncForgersReset();
         return count;
     }
 
+    private void scheduleAsyncForgersReset() {
+        if (!propertiesHolder.isLightClient() && generateBlocksTask != null) {
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:147]: light client check would be sufficient here. Also i'd suggest to change method name for generateBlocksTask.setConditionToResetForgers into 'resetForgersLater' or 'resetForgersAsync' or smth like that

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java [hunk 4]
```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
...
-            globalSync.updateLock();
-            try {
-                generateBlocksTask.resetSortedForgers();
-            } finally {
-                globalSync.updateUnlock();
-            }
-            log.debug(generator + " stopped ({} ms)", (System.currentTimeMillis() - start));
+        if (generator != null) {
+            log.debug(generator + ": Stop Forging command added to the queue.");
+            scheduleAsyncForgersReset();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:127]: Probably parameter injection into a string pattern with '{} my log string' would be more consistent approach

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java (hunk 3 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:147]: renamed

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java (hunk 4 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java:127]: fixed

**Review by AndrewBoyarsky — APPROVED** [review]
Looks ok

**Pull request merged**